standard/statements.md

```diff
@@ -351,11 +351,72 @@ It is an error to refer to a local variable by name in a textual position that p
 
 The ref-safe-context ([§9.7.2](variables.md#972-ref-safe-contexts)) of a ref local variable is the ref-safe-context of its initializing *variable_reference*. The ref-safe-context of non-ref local variables is *declaration-block*.
 
+When *local_variable_declaration* includes `using`, the semantics described above are modified, as follows.
+
+Each *local_variable_type* shall be `dynamic` or a resource type. If `await` is also present, the resource type shall not be a ref struct.
+
+Each *local_variable_declarator* shall have a *local_variable_initializer*.
+
+The variables declared are read-only. A compile-time error occurs if any subsequent statement attempts to modify these variables, take the address of them, or pass them as reference or output parameters.
+
+An occurrence of *local_variable_declaration* containing `using` has the same semantics as, and can be rewritten as, the corresponding resource-acquisition form of the using statement ([§13.14](statements.md#1314-the-using-statement)), as follows:
+
+```csharp
+using «local_variable_type» «local_variable_declarators»
+    // statements
+```
+
+is equivalent to
+
+```csharp
+using («local_variable_type» «local_variable_declarators»)
+{
+    // statements
+}
+```
+
+and
+
+```csharp
+await using «local_variable_type» «local_variable_declarators»
+    // statements
+```
+
+is equivalent to
+
+```csharp
+await using («local_variable_type» «local_variable_declarators»)
+{
+    // statements
+}
+```
+
+The lifetime of the variables declared in a *local_variable_declaration* extends to the end of the scope in which they are declared. Those variables are then disposed in the reverse order in which they are declared.
+
+<!-- Example: {template:"code-in-partial-class", name:"LocalVariableDecls6", additionalFiles:["SupportLocalVarDecl.cs"], replaceEllipsis:true, customEllipsisReplacements: ["\"File1.txt\", FileMode.Create", "\"File2.txt\", FileMode.Create", "\"File3.txt\", FileMode.Create"]} -->
+```csharp
+static void M()
+{
+    using FileStream f1 = new FileStream(...);
+    using FileStream f2 = new FileStream(...), f3 = new FileStream(...);
+    ...
+    // Dispose f3
+    // Dispose f2 
+    // Dispose f1
+}
+```
+
+A `goto` cannot jump to a location before a `using` declaration within the same block.
```

Comment:
I think this is incorrect, the point is you cannot jump over using in either direction. Basing wording from the [LDM notes](https://github.com/dotnet/csharplang/blob/3c8559f186d4c5df5d1299b0eaa4e139ae130ab6/meetings/2018/LDM-2018-12-03.md#goto-and-using-declarations) I’ll offer:

```suggestion
When jumping to a label in a statement list which contains *local_variable_declaration*s involving `using`, there must not be any such declaration between the `goto` statement and the label.```

Apart from wordsmithing this needs x-ref to goto, and goto probably needs this restriction listed in its description as well.

Also surely this is an echo of a similar statement for `using (…) {…}`, so is it needed here at all given the equivalency defined above – this is just a shorthand.